web/packages/core/src/ruffle-player.ts

```diff
@@ -1475,6 +1475,65 @@ export function isBuiltInContextMenuVisible(menu: string | null): boolean {
     return false;
 }
 
+/**
+ * Returns whether the given filename is a Youtube Flash source.
+ *
+ * @param filename The filename to test.
+ * @returns True if the filename is a Youtube Flash source.
+ */
+export function isYoutubeFlashSource(filename: string | null): boolean {
+    if (filename) {
+        let pathname = "";
+        let hostname = "";
+        try {
+            // A base URL is required if `filename` is a relative URL, but we don't need to detect the real URL origin.
+            const url = new URL(filename, RUFFLE_ORIGIN);
+            pathname = url.pathname;
+            hostname = url.hostname;
+        } catch (err) {
+            // Some invalid filenames, like `///`, could raise a TypeError. Let's fail silently in this situation.
+        }
+        // See https://wiki.mozilla.org/QA/Youtube_Embedded_Rewrite
+        if (
+            pathname.startsWith("/v/") &&
+            (hostname.endsWith("youtube.com") ||
+                hostname.endsWith("youtube-nocookie.com"))
```

Comment:
A little paranoid, but this will match with `notreallyyoutube.com/v/`.

Reply:
First commit of two new ones addressed this. The second one makes things work on Firefox the same way they work on Chrome, despite a discovered browser bug.

Reply:
Second new commit is to make this work on Firefox as well: https://github.com/ruffle-rs/ruffle/pull/6553#issuecomment-1086958924